### PR TITLE
Parse the test cases results log by json

### PR DESCRIPTION
**Pull request by wanghongjuan**
- Generate a json file to collect the test results.
- Locate easily the failed tests in result file when using the results file
- Quickly generate results to match the specified format that QA required
- JSON dose not need to take up extra workspace.

Signed-off-by: Wang Hongjuan hongjuan.wang@intel.com

### CODE REVIEW

**Comment by wanghongjuan** [comment]
@gabrielschulhof Could you please review it? Thanks.

File: bindings/nodejs/tests/getresult.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  caseList = [],
+  setList = [],
+  allList = [],
+  test = {},
+  testInfo = {};
+
+exports.getTestResult = function( status, success, failure  ) {
+
+  if ( status.result ) {
+    success = "PASS";
```
Comment (edersondisouza) [inline bindings/nodejs/tests/getresult.js]: Is this right? It appears that you get `success` and `failure` as parameters and change their values. But the caller already set a value for them, why do you do that?

File: bindings/nodejs/tests/setup.js [hunk 2]
```diff
@@ -56,6 +57,8 @@ QUnit.config.callbacks.log.push( function( status ) {
 
 	// Parameters: status: { module, result(t/f), message, actual, expected, testId, runtime }
 
+	results.getTestResult( status, success, failure );
```
Comment (edersondisouza) [inline bindings/nodejs/tests/setup.js]: Here it appears that `success` and `failure` will have their value changed by this call. Is this right?

File: bindings/nodejs/tests/getresult.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  } else if ( !status.result ) {
+    failure = "FAIL";
+  }
+
+  testInfo = {
+    "message": (  status.runtime  + ": " + status.message ),
+    "result": ( status.result ? success : failure ),
```
Comment (edersondisouza) [inline bindings/nodejs/tests/getresult.js]: Maybe all that you wanted here was something like:
```js
"result": ( status.result ? "PASS" : "FAIL" ),
```
With no need for `success` and `failure` parameters?

Comment (wanghongjuan) [inline bindings/nodejs/tests/getresult.js]: Many thanks for propose a good way, I will update my code.

**Comment by wanghongjuan** [comment]
@edersondisouza I create a new PR https://github.com/solettaproject/soletta/pull/1921, please review it. Thanks.

**Pull request closed without merging**